guides/05_chatbots/03_agents-and-tool-usage.md

```diff
@@ -35,25 +35,24 @@ We'll start by importing the necessary classes from transformers and gradio.
 ```python
 import gradio as gr
 from gradio import ChatMessage
-from transformers import load_tool, ReactCodeAgent, HfEngine
-from utils import stream_from_transformers_agent
+from transformers import load_tool, ReactCodeAgent # type: ignore
+from transformers.agents import stream_to_gradio, HfApiEngine # type: ignore
 
 # Import tool from Hub
-image_generation_tool = load_tool("m-ric/text-to-image")
+image_generation_tool = load_tool("huggingface-tools/text-to-image")
 
-
-llm_engine = HfEngine("meta-llama/Meta-Llama-3-70B-Instruct")
-# Initialize the agent with both tools
+llm_engine = HfApiEngine("meta-llama/Meta-Llama-3-70B-Instruct")
```

Comment:
There are some discrepancies between the code here and the demo above, e.g. the llm engine above is `Qwen/Qwen2.5-Coder-32B-Instruct`

Reply:
Yes good point!